### ISSUE TITLE
Multiple false positives with PHP 8.4 property definitions

### ISSUE THREAD

**Issue by tminich** [kalessil/phpinspectionsea#1970]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 5.1.0 |
| Language level | PHP 8.4                        |

Current behaviour
--
*Shortened asymmetric access declaration triggers `should be declared with an access modifier`*

![Image](https://github.com/user-attachments/assets/f6db71f0-8bd9-4a15-9699-a3abe61cbdb4)
While possible to add `public`, it's not necessary and default PHPStorm inspections recommend removing it.

*Getter and setter trigger `X can be declared as a return hint`*

![Image](https://github.com/user-attachments/assets/e447e594-2cd3-4098-af67-e52b00d890e7)
This is completely off as it's not valid php syntax.

*`Unnecessary final modifier` on getter and setter*
Visible in the previous screenshot. I have no idea where this comes from, probably triggered by something else in the class as it is not in a simplified test case.

Expected behaviour
--
None of these should trigger

Environment details
---

![Image](https://github.com/user-attachments/assets/b900994c-7309-4ffc-b07e-52e81690deec)

**Comment by Simbiat** [comment]
Just in case, same behavior in PHPStorm 2025.1 with EA Ultimate 2024.3
```
PhpStorm 2025.1
Build #PS-251.23774.436, built on April 14, 2025
Source revision: effa99765bfff
Licensed to Dmitrii Kustov
Subscription is active until March 3, 2026.
Runtime version: 21.0.6+9-b895.109 amd64 (JCEF 122.1.9)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.awt.windows.WToolkit
Windows 11.0
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 5120M
Cores: 16
Registry:
  debugger.new.tool.window.layout=true
  run.processes.with.pty=TRUE
  ide.experimental.ui=true
  terminal.new.ui=true
Non-Bundled Plugins:
  com.intellij.ml.llm (251.23774.42.28)
  de.espend.idea.php.annotation (12.0.0)
  fr.adrienbrault.idea.symfony2plugin (2024.1.276)
  com.kalessil.phpStorm.phpInspectionsUltimate-mp (2024.3)
```

**Comment by Simbiat** [comment]
I know, but original issue was reported for EA Extended 5.1.0, and I am using Ultimate, so I thought it would be worth it to mention that this is present in both.

**Comment by apuckey** [comment]
I am also getting this:
<img width="703" height="100" alt="Image" src="https://github.com/user-attachments/assets/9e8562b2-d3c2-4c21-b434-dcb4d68b92a4" />

**Comment by jdreesen** [comment]
@ea-inspections-team is there any plan to support the latest PHP features?

**Comment by kalessil** [comment]
As far as i aware should be released by end of September.

**Comment by jdreesen** [comment]
Thx for the update. A little more communication from the development team would be lovely.

**Comment by ea-inspections-team** [comment]
@kalessil : huh, you probably misunderstood something - the end of September is when return from my extended leave.

But I'm looking into this and another high-priority issues already. Let's aim next Wednesday for the release (uploading to the marketplace).

**Comment by kalessil** [comment]
@ea-inspections-team : sorry, my bad - I misunderstood indeed. And thanks for jumping in right away!

**Comment by ea-inspections-team** [comment]
@jdreesen : thank you for the ping. Yes, taking #1979 and #1978 into work for the next release (ETA next Wednesday).

Last releases had to cover many fixes for IDEs compatibility and postponed the work on features support.
Frustrating and unpredictable amount of work, but it has to be done to comply with marketplace internals.
